### PR TITLE
Solution

### PR DESCRIPTION
**Pull request by DmytroBondariev**

### CODE REVIEW

File: app/main.py [hunk 1]
```diff
@@ -1,3 +1,57 @@
+from __future__ import annotations
+
+
 class Distance:
-    # Write your code here
-    pass
+    def __init__(self, km: int) -> None:
```
Comment (Vasyl-Smutok) [inline app/main.py:5]: `float` possible

File: app/main.py [hunk 2]
```diff
@@ -1,3 +1,57 @@
...
+        if isinstance(other, Distance):
+            return Distance(km=self.km + other.km)
+        return Distance(km=self.km + other)
+
+    def __iadd__(self, other: Distance | int | float) -> Distance:
+        if isinstance(other, Distance):
+            self.km += other.km
+            return self
+        self.km += other
+        return self
```
Comment (Vasyl-Smutok) [inline app/main.py:24]: `DRY`
22 and 24 lines

File: app/main.py [hunk 3]
```diff
@@ -1,3 +1,57 @@
...
+
+    def __mul__(self, other: int) -> Distance:
+        return Distance(
+            km=self.km * other
+        )
```
Comment (Vasyl-Smutok) [inline app/main.py:29]:
```suggestion
        return Distance(km=self.km * other)
```

**Review by Vasyl-Smutok — CHANGES_REQUESTED** [review]

File: app/main.py (hunk 2 above)
Comment (Dmytry95) [reply app/main.py:24]: if i make method look like this, i don`t pass several tests   
```
def __iadd__(self, other: Distance | int | float) -> Distance:
        if isinstance(other, Distance):
            self.km += other.km
        self.km += other
        return self
```

**Review by Grayder0152 — APPROVED** [review]

**Pull request closed without merging**